util/image_viewer/spotfinder_frame.py

```diff
@@ -96,8 +96,51 @@ def __init__(self, *args, **kwds):
 
         super().__init__(*args, **kwds)
 
+        self.viewing_stills = True
+        for explst in self.experiments:
+            for exp in explst:
+                if exp.scan is not None or exp.goniometer is not None:
+                    self.viewing_stills = False
+                    break
+
+        if self.viewing_stills:
+            is_multi_shot_exp = [len(El) > 1 for El in self.experiments]
+            if any(is_multi_shot_exp):
```

Comment:
short-circuit evaluation and lower case loop variable
```suggestion
            is_multi_shot_exp = any(len(exp_list) > 1 for exp_list in self.experiments)
            if is_multi_shot_exp:
```